### PR TITLE
Customer Request #224686 [BE] Disability update api validations added

### PR DESCRIPTION
**Pull request by GitVivekHub**
https://tracker.tekdi.net/issues/224686

# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

## Summary by CodeRabbit

- **New Features**
  - Enhanced data handling for beneficiaries, particularly regarding disability status.
  - Introduced state-specific validation for government advantages based on program ID.

- **Improvements**
  - Streamlined response structure for beneficiaries without disabilities.
  - Refined validation logic to ensure efficiency and accuracy based on the `has_disability` flag.

### CODE REVIEW

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 1]
```diff
@@ -4280,27 +4303,278 @@ export class BeneficiariesService {
...
+				errors = setErrors(
+					errors,
+					'disability_occurence',
+					'Invalid disability_occurence data',
+				);
+			}
+		}
+
+		if (Object.keys(errors).length > 0) {
+			return response.status(400).json({
```
Comment (sagarkoshti1990) [inline src/src/beneficiaries/beneficiaries.service.ts:4571]: The expected status code will be 422.

File: src/src/beneficiaries/beneficiaries.service.ts [hunk 2]
```diff
@@ -4280,27 +4303,278 @@ export class BeneficiariesService {
...
+						...(errors?.[key]?.__errors || []),
+						`${message}`,
+					],
+				},
+			};
+		};
+
+		if (!body?.has_disability) {
+			missing_fields.push('has_disability');
+			return response.status(400).json({
```
Comment (sagarkoshti1990) [inline src/src/beneficiaries/beneficiaries.service.ts:4323]: The expected status code will be 422.

**Review by sagarkoshti1990 — CHANGES_REQUESTED** [review]

File: src/src/beneficiaries/beneficiaries.service.ts (hunk 1 above)
Comment (GitVivekHub) [reply src/src/beneficiaries/beneficiaries.service.ts:4571]: resolved

File: src/src/beneficiaries/beneficiaries.service.ts (hunk 2 above)
Comment (GitVivekHub) [reply src/src/beneficiaries/beneficiaries.service.ts:4323]: resolved

**Pull request merged**